tests/unit/test.program.js

```diff
@@ -430,6 +435,205 @@ describe('program.main', () => {
       assert.strictEqual(options.shouldExitProgram, false);
     });
   });
+
+  it('applies options from the specified config file', () => {
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const resolvedFakePath = path.resolve(fakePath);
+    const fakeLoadJSConfigFile = sinon.spy(() => {});
+    const fakeApplyConfigToArgv = sinon.spy(() => {});
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(() => {
+        const options = fakeCommands.lint.firstCall.args[0];
+        assert.strictEqual(options.config, fakePath);
+        sinon.assert.calledOnce(fakeLoadJSConfigFile);
+        sinon.assert.calledWith(fakeLoadJSConfigFile,
+                                sinon.match(resolvedFakePath));
+        sinon.assert.calledOnce(fakeApplyConfigToArgv);
+        sinon.assert.calledWith(fakeApplyConfigToArgv, sinon.match({
+          configFileName: resolvedFakePath,
+        }));
+      });
+  });
+
+  it('throws a UsageError when the config file can\'t be loaded', () => {
```

Comment:
I don't think you need this test because the behavior of handling thrown errors is already well defined. I can see that perhaps you wanted to be sure that the config checking code is executed from within the catch block but I don't think it's necessary to cover that logic. If one day we encounter a bug where someone accidentally moved it outside of the catch block then we could add a test for it. Until then, I think code review is a better way to make sure the config logic is executed with the catch block.